src/core/shape_inference/include/element_visitor.hpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <functional>
+
+#include "openvino/core/except.hpp"
+#include "openvino/core/type/element_type.hpp"
+
+namespace ov {
+namespace element {
+
+/**
+ * @brief Defines supported elements for applying the element visitor action.
+ *
+ * @tparam List of supported ov::element types.
+ *
+ * The apply function check input element type, if is on the list apply function of visitor for specific type
```

Comment:
Does it regard only input elements? Or may it be used for just element type?

Reply:
Comment updated.